enhancements/microshift/microshift-telemetry.md

```diff
@@ -0,0 +1,293 @@
+configuration option shall be added. This option will drive how often a metrics
+payload should be sent and it will default to 24h. In every MicroShift start
+all metrics shall be sent. Afterwards, it will follow the configuration option
+to schedule the next send.
+
+As described above MicroShift will be using the [direct request](https://github.com/openshift/telemeter/tree/main?tab=readme-ov-file#metricsv1receive-endpoint-receive-metrics-in-prompbwriterequest-format-from-any-client)
+endpoint.
+Each metric must follow [Prometheus WriteRequest](https://github.com/prometheus/prometheus/blob/release-2.38/prompb/remote.proto#L22)
+format.
+
+### Sampling and batching metrics
+Metrics sent by MicroShift can be categorized in:
+* Static metrics. These are fixed throughout the execution of MicroShift, such
+as memory or CPU.
+* Dynamic metrics. These change and evolve throughout the execution of
+MicroShift, such as resource usage or resources in the cluster.
+
+If we assume default values, metrics are sent at least once a day in those
+deployments that do not disable the functionality. While this is good enough
+for static metrics, it provides a degraded view for dynamic ones. Having
+once-per-day data on resource usage makes it virtually impossible to extract
+any patterns out of it. For this reason it might be beneficial to sample
+dynamic metrics more often and then batch them together with the static
+metrics when sending them.
+
+Sampling metrics means MicroShift needs to store values in between metrics
+reports. The dynamic metrics are not numerous enough to take a toll on resource
+usage, but this interval must be configurable and default to a sensible value.
+
+As seen before each metric is under 250B. Considering 7 metrics, having one of
+them hold multiple samples (one for each resource type, totalling ) a complete
+metrics payload with static and dynamic metrics would be under 2 KB.
+Using defaults (1 payload every 24h, with 1h samples for all dynamic metrics)
+would be under 40 KB.
+
+### Sensitive data
+There is no user or private data in any of the metrics in MicroShift reports.
+
+### Opting out
+MicroShift may not always have the possibility of sending metrics. There may be
+disconnected clusters, constrained environments where external traffic could be
+audited, customers who simply do not want to share this information, etc. For
+these reasons there must be an opt-out for this functionality, as OpenShift
+already [allows](https://docs.openshift.com/container-platform/4.17/support/remote_health_monitoring/opting-out-of-remote-health-reporting.html).
+
+Taking advantage of the configuration options that the feature requires, an
+enable/disable toggle is provided.
+
+### Workflow Description
+**MicroShift** is the MicroShift main process.
+
+1. MicroShift starts up.
+2. MicroShift reads configuration. If telemetry is not enabled, finish here. If telemetry is enabled proceed to next step.
```

Comment:
OK, so it seems like we want to configure this at install, prior to first boot. Can we turn this on later and restart? (And vice versa, turn it off and restart.)

Reply:
Yes you can enable/disable this everytime you restart MicroShift, it is not tied to the install/first boot.